### ISSUE TITLE
Unable to find OverlayLayer from given TopLevel

### ISSUE THREAD

**Issue by BuLuoPiaoYu** [amwx/FluentAvalonia#596]
**Describe the bug**
When I create a MainWindow form and inherit from my custom form class AvaWindow, calling ContentDialog.ShowAsync() within it raises the following exception: "Unable to find OverlayLayer from given TopLevel".

**Screenshots**
![image](https://github.com/user-attachments/assets/0596f448-a8e2-429f-b625-c0fd03c46b04)

**Desktop/Platform (please complete the following information):**
 - OS: Windows 11
 - FluentAvalonia Version [2.1.0-preview6]
 - Avalonia Version [11.1.0]
 
**Here is the code**
[Style and Code.zip](https://github.com/user-attachments/files/16343120/Style.and.Code.zip)

**Comment by salty-flower** [comment]
I got a similar exception on Android when tapping on `FAComboBox`:
> Unable to create IPopupImpl and no overlay layer is found for the target control

Test with many combinations and the last working combination is 
- `Avalonia(.*)`: 11.1.0-beta1
- `FluentAvalonia`: 2.1.0-preview5

Bumping either will bring the exception for me.

Looking at [changes made between Avalonia `11.1.0-beta1` and `11.1.0-beta2`](https://github.com/AvaloniaUI/Avalonia/compare/11.1.0-beta1...11.1.0-beta2), I noticed [this commit](https://github.com/AvaloniaUI/Avalonia/commit/f18bf085d905d0997037aceb170b5ff3ad7ecf49) has changed Android `TopLevel` significantly. Not sure if this is related.
